src/environment/game.py

```diff
@@ -15,15 +15,20 @@ def __init__(self, initial_game_state: GameState, visualize: bool = True):
         if visualize:
             GridVisualizer.visualize(self._game_state)
 
-    def play(self, instruction_text: str) -> None:       
-        instructions = PlayerActionParser.parse_raw(instruction_text)
-        for instruction in instructions:
-            action = PlayerActionParser.parse(instruction)
-            if action is not None:
-                action.act(self._game_state)
-            else:
-                # TODO: maybe write something better here?
-                print("Invalid instruction: " + instruction)
+    def play(self) -> None:
+        print(self._game_state.current_room.description)
+        while True:    
+            instruction = input("Enter your instruction: ")   
+            action = ActionParser.parse(instruction)
+            result = action.act(self._game_state)
+            if result.terminal:
+                if result.win:
+                    print("You win!")
+                 
+                else:
+                    print("You lose!")
+                return
```

Comment:
Nice! Pekny loop 👍🏼 

Ked sa na to pozeram, tak vidim alternativny pristup:

```py
if result.terminal:
    print("You win!" if result.win else "You lose!")
    return
```

kde zredukujes pocet riadkov if-else zo 4 na 1. Ber to len ako friendly suggestion, bit more concise.

Reply:
oki 🙂